### PR TITLE
Refactor/migrate vertex components to FC

### PR DESCRIPTION
**Pull request by bluessista**
This PR migrates the PolygonVertex and EdgeVertex class components to FC components

### CODE REVIEW

File: src/PolygonDraw/PolygonVertex.tsx [hunk 1]
```diff
@@ -21,93 +19,99 @@ export interface Props {
...
-    };
+    let circleMarkerElement: CircleMarker;
 
-    componentDidMount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.on('dragstart', this.handleDragStart);
-            this.circleMarkerElement.on('drag', this.handleDrag);
-            this.circleMarkerElement.on('dragend', this.handleDragEnd);
+    useEffect(() => {
+        if (!isDragged && !!coordinate) {
```
Comment (nlopin) [inline src/PolygonDraw/PolygonVertex.tsx:38]: do we need to check for existance of the `coordinate` here? Props say it is a required property which can only have coordinates object. If I'm not wrong, it means `!!coordinate` is always `true`

File: src/PolygonDraw/PolygonVertex.tsx [hunk 2]
```diff
@@ -21,93 +19,99 @@ export interface Props {
...
+    const handleDragEnd = useCallback(() => {
+        onDragEnd();
+        setIsDragged(false);
+    }, [onDragEnd]);
 
-    handleDragStart = (event: DragEvent) => {
-        this.props.onDragStart(event.target.getLatLng(), this.props.index);
-        this.setState({ isDragged: true });
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const setCircleMarkerRef = (ref: any) => {
```
Comment (nlopin) [inline src/PolygonDraw/PolygonVertex.tsx:66]: Could you help me to understand what is going on in this method? I see we have `cicrleMarkerElement` variable which is created on every render, and we assign something unknown which is called `ref`. Why are we doing this?

File: src/PolygonDraw/PolygonVertex.tsx (hunk 1 above)
Comment (bluessista) [reply src/PolygonDraw/PolygonVertex.tsx:38]: You are right, the additional check is not needed. Will be removed

File: src/PolygonDraw/PolygonVertex.tsx (hunk 2 above)
Comment (bluessista) [reply src/PolygonDraw/PolygonVertex.tsx:66]: Yes this one seems outdated and can cause bugs - I will give it another try using useRef and a different setter approach.

Comment (bluessista) [reply src/PolygonDraw/PolygonVertex.tsx]: I added a fix in this commit: #[fcf9656](https://github.com/freenowtech/react-polygon-editor/pull/27/commits/fcf96560e8d151c524eaf47654efc80e2a7f9a33)

File: src/PolygonDraw/PolygonVertex.tsx (hunk 1 above)
Comment (bluessista) [reply src/PolygonDraw/PolygonVertex.tsx]: I've added a fix in this commit: #[fcf9656](https://github.com/freenowtech/react-polygon-editor/pull/27/commits/fcf96560e8d151c524eaf47654efc80e2a7f9a33)

File: src/PolygonDraw/PolygonVertex.tsx [hunk 3]
```diff
@@ -21,93 +19,97 @@ export interface Props {
...
         }
-    }
+    }, []);
 
-    componentWillUnmount() {
-        if (this.circleMarkerElement) {
-            this.circleMarkerElement.off('dragstart', this.handleDragStart);
-            this.circleMarkerElement.off('drag', this.handleDrag);
-            this.circleMarkerElement.off('dragend', this.handleDragEnd);
+    useEffect(() => {
```
Comment (zackjones11) [inline src/PolygonDraw/PolygonVertex.tsx:43]: (optional, personal preference) i think we could remove this `useEffect` by adding `createLeafletLatLngFromCoordinate` as the state init value and in the `handleDragEnd` method

**Review by zackjones11 — APPROVED** [review]
😍🚀 great job!

File: src/PolygonDraw/PolygonVertex.tsx [hunk 4]
```diff
@@ -21,93 +19,97 @@ export interface Props {
...
+    onDragStart,
+}) => {
+    const [isHovered, setIsHovered] = useState(false);
+    const [isDragged, setIsDragged] = useState(false);
+    const [latLng, setLatLng] = useState(new LatLng(0, 0));
 
-    static getDerivedStateFromProps = (props: Props, state: State) => {
-        return state.isDragged ? state : { ...state, latLng: createLeafletLatLngFromCoordinate(props.coordinate) };
-    };
+    const circleMarkerRef = useRef<CircleMarker>();
```
[the remark this answers is not in the document]
Comment (bluessista) [reply src/PolygonDraw/PolygonVertex.tsx]: thank you - I implemented it and it works like a charm

File: src/PolygonDraw/PolygonVertex.tsx (hunk 3 above)
Comment (bluessista) [reply src/PolygonDraw/PolygonVertex.tsx:37]: This is unfortunately not working as the coordinates need to stay as reactive dependency that the marker needs to know about. Without the effect the marker stays at an old position e.g. when the undo action is used.

**Review by nlopin — APPROVED** [review]

**Pull request merged**